packages/react/src/components/Dialog/Dialog.tsx

```diff
@@ -70,22 +70,22 @@ export interface DialogProps
 
   /**
    * Called after a user clicks the cancel button.
-   * @param {SyntheticEvent} event - React's original SyntheticEvent.
-   * @param {object} data - All props.
+   * @param event - React's original SyntheticEvent.
+   * @param data - All props.
```

Comment:
The changes here in the JSDoc will affect the TODO in `docs/src/components/ComponentDoc/ComponentProp/ComponentPropFunctionSignature.tsx`:

```jsx
        {/* TODO change these according to the react-docgen-typescript generated json */}
        {/* <ComponentPropFunctionSignature name={name} tags={tags} /> */}
```

This doc component shows the function signature for a callback in the prop table, like so:

![image](https://user-images.githubusercontent.com/5067638/69284532-b572dd00-0ba3-11ea-83b2-3d6e5c4d9b62.png)
![image](https://user-images.githubusercontent.com/5067638/69284586-d0dde800-0ba3-11ea-90b3-4ce341b2ceb0.png)

It is currently disabled because the types (`SyntheticEvent`, `object`) are not currently getting parsed.  They are null.

***

I think this means moving forward we'll need to rely on actual typing resolution to figure out what the signature of `ComponentEventHandler<>` is.  Correct?